### PR TITLE
feat: [PL-27160]: Replace account settings with settings framework

### PR DESCRIPTION
**Pull request by tiwaritejas**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38761)

### CODE REVIEW

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 12 of 13 files at r1, 2 of 2 files at r2, 3 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38761)* status: 15 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java` line 242 at r3](https://reviewable.io/reviews/harness/harness-core/38761#-NEGq-VwC1AVK39R8vu-:-NEGq-VwC1AVK39R8vu0:b-yo7l6h) ([raw file](https://github.com/harness/harness-core/blob/cdf4fe6c0fb75aabf80eeb4ea83e276df9eeeead/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java#L242)):*
> ```Java
>       String featureFlagErrorMessage =
>           String.format("Invalid request: Feature not available for your account- %s", accountIdentifier);
>       if (!featureFlagErrorMessage.equals(ex.getMessage())) {
> ```

The call should be made based on FF value. Same every where else.

File: 120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java [hunk 1]
```diff
@@ -227,11 +228,25 @@ public SecretResponseWrapper create(String accountIdentifier, SecretDTOV2 dto) {
...
+          String.format("Invalid request: Feature not available for your account- %s", accountIdentifier);
+      if (!featureFlagErrorMessage.equals(ex.getMessage())) {
+        throw ex;
+      }
+    }
 
     if (isBuiltInSMDisabled && isHarnessManaged) {
-      throw new InvalidRequestException("Built-in Harness Secret Manager cannot be used to create Secret.");
+      throw new InvalidRequestException(
+          "Built-in Harness Secret Manager cannot be used to create Secret as it has been disabled.");
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java:249]: Inform QE for change of error msg.

File: 440-connector-nextgen/src/main/java/io/harness/connector/impl/ConnectorStatisticsHelper.java [hunk 2]
```diff
@@ -74,8 +77,22 @@ private Criteria createCriteriaObjectForConnectorScope(
...
             .in(projectIdentifier)
             .orOperator(where(ConnectorKeys.deleted).exists(false), where(ConnectorKeys.deleted).is(false));
-    boolean isBuiltInSMDisabled =
-        accountSettingService.getIsBuiltInSMDisabled(accountIdentifier, null, null, AccountSettingType.CONNECTOR);
+
+    Boolean isBuiltInSMDisabled = false;
+    try {
+      isBuiltInSMDisabled = parseBoolean(
+          NGRestUtils
+              .getResponse(settingsClient.getSetting(
```
Comment (meenakshi-harness) [inline 440-connector-nextgen/src/main/java/io/harness/connector/impl/ConnectorStatisticsHelper.java:85]: same at all places

File: 440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java [hunk 3]
```diff
@@ -226,9 +228,20 @@ private Page<Connector> listHelper(int page, int size, String accountIdentifier,
...
+    try {
+      isBuiltInSMDisabled = parseBoolean(
+          NGRestUtils
+              .getResponse(settingsClient.getSetting(
+                  SettingIdentifiers.DISABLE_HARNESS_BUILT_IN_SECRET_MANAGER, accountIdentifier, null, null))
+              .getValue());
+    } catch (InvalidRequestException ex) {
+      String featureFlagErrorMessage =
+          String.format("Invalid request: Feature not available for your account- %s", accountIdentifier);
+      if (!featureFlagErrorMessage.equals(ex.getMessage())) {
```
Comment (meenakshi-harness) [inline 440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java:241]: try to move in a common helper, as its used at multiple places

File: 440-connector-nextgen/src/test/java/io/harness/connector/impl/DefaultConnectorServiceImplTest.java [hunk 4]
```diff
@@ -122,6 +125,12 @@ public void setUp() throws Exception {
     passwordSecretRef = SecretRefData.builder().identifier(passwordIdentifier).scope(Scope.ACCOUNT).build();
     doNothing().when(secretRefInputValidationHelper).validateTheSecretInput(any(), any());
     doReturn(false).when(gitSyncSdkService).isGitSyncEnabled(anyString(), anyString(), anyString());
+    when(settingsClient.getSetting(
+             SettingIdentifiers.DISABLE_HARNESS_BUILT_IN_SECRET_MANAGER, accountIdentifier, null, null))
+        .thenReturn(request);
+    SettingValueResponseDTO settingValueResponseDTO =
+        SettingValueResponseDTO.builder().value("false").valueType(SettingValueType.BOOLEAN).build();
+    when(request.execute()).thenReturn(Response.success(ResponseDTO.newResponse(settingValueResponseDTO)));
```
Comment (meenakshi-harness) [inline 440-connector-nextgen/src/test/java/io/harness/connector/impl/DefaultConnectorServiceImplTest.java:133]: move to specific test only.

**Review by tiwaritejas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38761)* status: 13 file(s) to review. 1 change requests. 6. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java` line 237 at r3](https://reviewable.io/reviews/harness/harness-core/38761#-NEFObfuwphq0FqFhJnb-r3-237:-NFImAagEi1S6bq2DS5v:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/cdf4fe6c0fb75aabf80eeb4ea83e276df9eeeead/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java#L237)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

call this only when FF is on. Also don't need try catch here.
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java` line 242 at r3](https://reviewable.io/reviews/harness/harness-core/38761#-NEGq-VwC1AVK39R8vu-:-NFImB-vDDQ7ZJ266qZP:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/cdf4fe6c0fb75aabf80eeb4ea83e276df9eeeead/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java#L242)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

The call should be made based on FF value. Same every where else.
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java` line 249 at r3](https://reviewable.io/reviews/harness/harness-core/38761#-NEFObfuwphq0FqFhJnb-r3-249:-NFJAJ_h8K1OGZLn1xH0:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/cdf4fe6c0fb75aabf80eeb4ea83e276df9eeeead/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/SecretCrudServiceImpl.java#L249)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Inform QE for change of error msg.
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/ConnectorStatisticsHelper.java` line 85 at r3](https://reviewable.io/reviews/harness/harness-core/38761#-NEFObfuwphq0FqFhJnh-r3-85:-NFImDBe1voOPxaKAGLB:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/cdf4fe6c0fb75aabf80eeb4ea83e276df9eeeead/440-connector-nextgen/src/main/java/io/harness/connector/impl/ConnectorStatisticsHelper.java#L85)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

same at all places
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 241 at r3](https://reviewable.io/reviews/harness/harness-core/38761#-NEFObfuwphq0FqFhJni-r3-241:-NFJ8mPk9ThpiPzrNqRH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/cdf4fe6c0fb75aabf80eeb4ea83e276df9eeeead/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L241)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

try to move in a common helper, as its used at multiple places
</blockquote></details>

Done.
___
*[`440-connector-nextgen/src/test/java/io/harness/connector/impl/DefaultConnectorServiceImplTest.java` line 133 at r3](https://reviewable.io/reviews/harness/harness-core/38761#-NEFObfuwphq0FqFhJnl-r3-133:-NFJAHah7xKyjodf9tBW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/cdf4fe6c0fb75aabf80eeb4ea83e276df9eeeead/440-connector-nextgen/src/test/java/io/harness/connector/impl/DefaultConnectorServiceImplTest.java#L133)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

move to specific test only.
</blockquote></details>

Done.

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 7 of 9 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38761)* status: 13 file(s) to review. 1 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`440-connector-nextgen/BUILD.bazel` line 113 at r4](https://reviewable.io/reviews/harness/harness-core/38761#-NFMXEep6rFoxflY7qJ9:-NFMXEeq82K92OYd-j8Q:bcgjvre) ([raw file](https://github.com/harness/harness-core/blob/3f97c648a5c57b64ef6d54d2069c70ba8af275da/440-connector-nextgen/BUILD.bazel#L113)):*
> ```Starlark
>         "@maven//:org_assertj_assertj_core",
>         "@maven//:org_hibernate_validator_hibernate_validator",
>         "@maven//:org_mockito_mockito_inline",
> ```

Why this change?
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 1168 at r4](https://reviewable.io/reviews/harness/harness-core/38761#-NFMXpb-0qHLYmVFCJGc:-NFMXpb-0qHLYmVFCJGd:bpjomyb) ([raw file](https://github.com/harness/harness-core/blob/3f97c648a5c57b64ef6d54d2069c70ba8af275da/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L1168)):*
> ```Java
>   }
> 
>   public Boolean isBuiltInSMDisabled(String accountIdentifier) {
> ```

Shouldn't this be a private?

**Review by tiwaritejas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38761)* status: 13 file(s) to review. 1 change requests. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`440-connector-nextgen/BUILD.bazel` line 113 at r4](https://reviewable.io/reviews/harness/harness-core/38761#-NFMXEep6rFoxflY7qJ9:-NFMx897Bte9-tnE7KaN:bb1q348) ([raw file](https://github.com/harness/harness-core/blob/3f97c648a5c57b64ef6d54d2069c70ba8af275da/440-connector-nextgen/BUILD.bazel#L113)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

Why this change?
</blockquote></details>

I wanted mockStatic() for some UTs, hence made this change. Should not break anything as all of mockito_core functions are available.
___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 1168 at r4](https://reviewable.io/reviews/harness/harness-core/38761#-NFMXpb-0qHLYmVFCJGc:-NFMxI-h4nm5CO7S95jw:b-m2u2b0) ([raw file](https://github.com/harness/harness-core/blob/3f97c648a5c57b64ef6d54d2069c70ba8af275da/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L1168)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

Shouldn't this be a private?
</blockquote></details>

Yes, made the change

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 1 of 9 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38761)* status: 13 file(s) to review. 5. unresolved discussions. 1 of 2 approvals obtained.

**Comment by tiwaritejas** [comment]
trigger checkstyle

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 1 of 13 files at r1, 1 of 2 files at r2, 3 of 7 files at r3, 8 of 9 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38761)* status: 1. unresolved discussions. 

___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 1168 at r5](https://reviewable.io/reviews/harness/harness-core/38761#-NFN4fY4D9r5NjLILo9O:-NFN4fY4D9r5NjLILo9P:bi0kaxn) ([raw file](https://github.com/harness/harness-core/blob/8f69389ca06c489165ad0ae04cac97548fe8a704/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L1168)):*
> ```Java
>   }
> 
>   private Boolean isBuiltInSMDisabled(String accountIdentifier) {
> ```

We can check later if we can move this to common class for reducing duplication

**Pull request merged**